### DISCUSSION TITLE
Can gnet support tun/tap interfaces?

### DISCUSSION THREAD

**Discussion by jantypas** [panjf2000/gnet#507] [Ideas]
I need a network packet interface -- tun or tap.   What would it take to get at those interfaces?

**Comment by panjf2000** [comment]
Sorry, I don't quite get your point, could you provide me more details?

**Comment by jantypas** [comment]
Allow me to elaborate -- I am trying to read directly from network interfaces rather than udp, tcp, etc.  These are ethernet frames  How/where do I use/extended gnet to do that?

**Reply by panjf2000** [reply]
`gnet` is a network framework that works on the transport layer, if I understood your question correctly, then there is no way to fulfill your requirement with `gnet`.
